### ISSUE TITLE
Build out Proposal Acceptance Guidelines

### ISSUE THREAD

**Issue by adammontville** [CISecurity/oval-governance-update#25]
Proposal Acceptance Guidelines are intended to provide the gist of the things that need to be considered with any proposal. This artifact will likely be referenced by at least one, but likely several others (i.e. PR instructions). This artifact may additionally reference other artifacts, such as the OVAL design principles.

**Comment by vanderpol** [comment]
**OVAL Language Update Proposal Guidelines**
Version: DRAFT

**1. Proposal Summary**
Write up a single sentence of what is to be updated in the OVAL Language.  This will be used for tracking your proposal from start to finish and may be used as part of the OVAL changelog if approved.

**2. Proposal Description**
Explain in detail, what is lacking in OVAL and what your proposal aims to accomplish.

**3. Backward Compatibility**
Document any backward compatibility issues this update may cause.

Note that breaking backwards compatibility will likely cause any proposal to be deferred until the next major version of the language, which could take years.

**4. New Capabilities**
Confirm that what you are proposing cannot be accomplished with existing OVAL.  If it can be accomplished with existing OVAL, but the proposed method is &#39;better&#39;, explain why.

**5. Complies with OVAL Design Principles**
Refer to &lt;OVAL Design Principles&gt; to ensure this proposal complies.

**6. Affected OVAL Schema Documents**
List the schema&#39;s to be updated, by name.  Example:

- windows-definitions-schema.xsd
- windows-system-characteristics-schema.xsd

**7. Document implementation**
Document how the data is gathered from the system, include any System API&#39;s, command line application used, parsing of configuration files etc..

While the implementation is not mandated for other OVAL vendors to follow, documenting which system calls were used to obtain the data will allow proposal reviewers to quickly understand what is being performed, and may allow other vendors to create their own prototypes.

**8. Documentation of API calls**
If any documentation exists from the Operating System or application vendor, please include any URLs below.

**9. OVAL Schema Update**
Obtain a copy of the most recent OVAL schema&#39;s available and update them based on your proposal.

**10. Sample Content**
Create verbose sample content, which demonstrates all capabilities of your proposal.  Creating content that tests all possible capabilities may also decrease the need for future updates after real world usage occurs.  The more detailed the content is, the higher the likelihood for approval and overall success.

Sample content must be able to be validated using the updated XML schema.

**11. Prototype**
Create a prototype application which performs the capabilities documented in your proposal, and processes the sample content created.

Creating a prototype demonstrates to the OVAL Board that the proposal is technically feasible and will help work out any issues.  If you are not an OVAL Interpreter developer, you may want to partner with an existing developer.

**12. Sample Results**
Include the OVAL results generated by your prototype and using the sample content listed in this proposal.

Results must be able to be validated using the updated XML schema.

**13. Bundling for OVAL Board review**
Create a zip file that contains:

A.  Your proposal based on this document
B. Subdirectories
  1. Update OVAL Schemas
  2. Sample Content
  3. Sample Results

**Comment by vanderpol** [comment]
Not sure if this is the best location to create the documentation, but figured it could be copied wherever it's needed.  Also pondering if creating OVAL governance documentation in the built in Wiki would make more sense?

**Comment by adammontville** [comment]
@vanderpol Thanks for posting this!  Let's review these at today's meeting, as I think this is the only thing that's been done since we all last met.

Also, I'm not opposed to using the wiki. The thinking behind using the markdown files in the `process_artifacts` directory was that we would more easily be able to port them over to the GitHub.io pages (I have no idea whether the wiki would make that equally straightforward).

**Comment by mpreisler** [comment]
This looks great. I am wondering whether `12.` shouldn't be in a patch format so that we can apply multiple proposals in parallel. A patch will also lower the scope for reviews.

**Comment by vanderpol** [comment]
I added the linking existing issue as step 1, per David's suggestion.  For creating a patch format, should we be linking to another page that explains this process?  Sounds fine, I'm just not a git person, and assume not all other proposal submitters will be either.

**Comment by adammontville** [comment]
Thanks for adding that, Jack. I'm not sure about where the patch information should exist. We can certainly create a separate artifact referenced by this one. Also, in the spirit of being more accepting than blocking, I think we should not necessarily require the patch format, but encourage it for the obvious reasons of space savings, etc.

What do others think about creating a separate process artifact relative to creating the submission packages using the method Martin suggested? 

If others are interested in this, Martin, would you be willing to author that artifact?

**Comment by DavidRies** [comment]
The patch format how-to and etc. could be a part of the GIT-oriented instructions we're planning on making to support each step of the process. I'd be happy to help create these, but it probably makes sense to wait until we're pretty happy with the whole end-to-end process first.

**Comment by adammontville** [comment]
That's a good idea, David.
